server/routes/charts.js

```diff
@@ -126,5 +144,49 @@ router
       return res.status(500).json({ message: error.message })
     }
   })
+  .get(ensureAuthenticated, async (req, res) => {
+    try {
+      const { dataDb } = req.app.locals
+      const chartForm = await dataDb
+        .collection(collections.charts)
+        .findOne({ _id: new ObjectID(req.params.chart_id) })
+      return res.status(200).json({ data: chartForm })
+    } catch (error) {
+      console.error(error)
 
+      return res.status(500).json({ message: error.message })
+    }
+  })
+  .put(ensureAuthenticated, async (req, res) => {
+    try {
+      const { dataDb } = req.app.locals
+      const { chart_id } = req.params
+      const { title, variable, assessment, description, fieldLabelValueMap } =
+        req.body
+      const { value, ok } = await dataDb
+        .collection(collections.charts)
+        .findOneAndUpdate(
+          { _id: new ObjectID(chart_id) },
+          {
+            $set: {
+              title,
+              variable,
+              assessment,
+              description,
+              fieldLabelValueMap,
+              updatedAt: new Date().toISOString(),
+            },
+          },
+          {
+            upsert: true,
+          }
+        )
+
+      return ok === 1
+        ? res.status(200).json({ data: value })
+        : res.status(400).json({ message: 'Chart information not found' })
```

Comment:
upsert true will insert a document if it can't find one to update, looking at this error message, do we want upsert true? alternatively should this message say something else? "error saving your changes" or something like that?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/118/commits/b0d022d6af9323225649d5f5e939b2d710b7b522